vhdbuilder/packer/install-dependencies.sh

```diff
@@ -195,28 +195,44 @@ if [[ $OS == $UBUNTU_OS_NAME ]]; then
   done
 fi
 
+if [[ $OS == $UBUNTU_OS_NAME && $(isARM64) != 1 ]]; then  # no ARM64 SKU with GPU now
+  if [[ "${CONTAINER_RUNTIME}" == "containerd" ]]; then
+    if grep -q "fullgpu" <<< "$FEATURE_FLAGS"; then
+      mkdir -p /opt/{actions,gpu}
+      ctr image pull docker.io/alexeldeib/aks-gpu:latest
+      ctr run --privileged --net-host --with-ns pid:/proc/1/ns/pid --mount type=bind,src=/opt/gpu,dst=/mnt/gpu,options=rbind --mount type=bind,src=/opt/actions,dst=/mnt/actions,options=rbind docker.io/alexeldeib/aks-gpu:latest gpuinstall /entrypoint.sh install.sh
+      ret=$?
+      if [[ "$ret" != "0" ]]; then
+        echo "Failed to install GPU driver, exiting..."
+        exit $ret
+      fi
+    else
+      mkdir -p /opt/{actions,gpu}
+      ctr image pull docker.io/alexeldeib/aks-gpu:latest
+      ctr run --privileged --net-host --with-ns pid:/proc/1/ns/pid --mount type=bind,src=/opt/gpu,dst=/mnt/gpu,options=rbind --mount type=bind,src=/opt/actions,dst=/mnt/actions,options=rbind docker.io/alexeldeib/aks-gpu:latest gpuinstall /entrypoint.sh copy
```

Comment:
just wondering whether this new approach works for FIPS+GPU?

currently for FIPS+GPU, at VHD build time, the build VM runs on non-fips kernel, but fips kernel is installed, so GPU driver built&installed are not for target fips kernel at this moment.

At AKS node provisioning time, the VM starts with FIPS kernel, the GPU driver is rebuilt and installed for the targeting fips kernel. In your new approach, will the gpu driver is built&installed again in CSE?

in cse_config.sh, the container image docker.io/alexeldeib/aks-gpu:latest is pulled and run, it looks FIPS+GPU should work as before, just want to confirm.

Reply:
this is also something we could improve for all our builds -- rebooting before build completion to get the latest kernel updates.

Reply:
ran a fips test build for gpu and seemed to work okay https://msazure.visualstudio.com/CloudNativeCompute/_build/results?buildId=58954783&view=logs&j=2e57e679-818b-5c8d-ac48-4ec72e0831ac